### PR TITLE
Add smoke tests

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

Will resolve https://github.com/paketo-buildpacks/samples/issues/47

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have added an integration test, if necessary.

### CODE REVIEW

**Comment by sophiewigmore** [comment]
Notes:
* .NET Core Runtime test fails with an [HTTP error](https://github.com/paketo-buildpacks/samples/pull/69/files#diff-abc681bf1e1527a253a214e7dc232266a5526993fc2c49b0d528cfbc5fa47757R88) when we try to check the http response body, so for this test we are only checking `BeAvailable()`
``
* Even when run sequentially, the Java/Java Native Image tests appear to fail when run within the entire suite

File: scripts/smoke.sh [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+      *)
+        util::print::error "unknown argument \"${1}\""
+    esac
+  done
+
+  if [[ ! -d "${SAMPLESDIR}/tests" ]]; then
+      util::print::warn "** WARNING  No Smoke tests **"
+  fi
+  # TODO: fix unary operator
```
Comment (ForestEckhardt) [inline scripts/smoke.sh:45]: Is this still to do?

File: tests/dotnet_test.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+					Expect(logs).To(ContainLines(ContainSubstring("Paketo .NET Core SDK Buildpack")))
+					Expect(logs).To(ContainLines(ContainSubstring("Paketo ICU Buildpack")))
+					Expect(logs).To(ContainLines(ContainSubstring("Paketo .NET Publish Buildpack")))
+					Expect(logs).To(ContainLines(ContainSubstring("Paketo .NET Execute Buildpack")))
+
+					Eventually(container).Should(BeAvailable())
+
+					// This is failing with 2021/02/09 16:06:56 Unsolicited response received on idle HTTP channel starting with "HTTP/1.1 200 OK
+					// TODO: look into this
+					// Eventually(container).Should(Serve(ContainSubstring("Powered By Paketo Buildpacks")).OnPort(8080))
```
Comment (ForestEckhardt) [inline tests/dotnet_test.go:90]: This should probably be looked into.

File: scripts/smoke.sh (hunk 1 above)
Comment (sophiewigmore) [reply scripts/smoke.sh:45]: Oops no it's not

File: tests/dotnet_test.go (hunk 2 above)
Comment (sophiewigmore) [reply tests/dotnet_test.go:90]: You and I saw this error before when testing .NET Core Runtime, where the test wasn't able to ping the container, but we were able to manually. The only "fix" was adding some extra code (like a sleep) between running the container and pinging it. Ryan thought this had to do with some kind of rate limiting. This seems kind of hacky.

**Review by kvedurmu — COMMENTED** [review]
This might be somewhat separate, but could we document in the README the recommended steps to test a new sample app? I'd expect folks outside of the core dev team to add sample apps in here, so it could be good to list what's recommended when adding a new app in.

**Comment by sophiewigmore** [comment]
@kvedurmu I like that idea. let me know if https://github.com/paketo-buildpacks/samples/pull/69/commits/d3216842f54d2bf91d08dff67c947c145b16abd8 covers that case well enough

File: tests/dotnet_test.go (hunk 2 above)
Comment (ForestEckhardt) [reply tests/dotnet_test.go:90]: Let me take a look at this.

**Comment by ForestEckhardt** [comment]
@sophiewigmore Could you add a `.gitignore` for the `.bin` folder that is create?

File: tests/dotnet_test.go (hunk 2 above)
Comment (ForestEckhardt) [reply tests/dotnet_test.go]: I was able to figure out this issue! Merged in the fix and included the Serve check.

**Review by ForestEckhardt — APPROVED** [review]

**Pull request merged**